projects/storefrontlib/src/layout/a11y/keyboard-focus/on-navigate/on-navigate-focus.service.ts

```diff
@@ -1,33 +1,56 @@
-import { DOCUMENT } from '@angular/common';
-import { Inject, Injectable } from '@angular/core';
+import { Inject, Injectable, OnDestroy } from '@angular/core';
 import { NavigationStart, Router } from '@angular/router';
 import { Subscription } from 'rxjs';
 import { filter, take } from 'rxjs/operators';
 import { BREAKPOINT } from '../../../config';
 import { BreakpointService } from '../../../../layout/breakpoint/breakpoint.service';
 import { KeyboardFocusConfig } from '../config';
+import { DOCUMENT } from '@angular/common';
 
 /**
  * Shared service for keyboard focus features called when the browser navigates.
  */
 @Injectable({
   providedIn: 'root',
 })
-export class OnNavigateFocusService {
+export class OnNavigateFocusService implements OnDestroy {
   protected resetFocusOnNavigate: Subscription;
   protected resetViewOnNavigate: Subscription;
 
+  /**
+   * @deprecated in 4.0
```

Comment:
IMHO we shouldn't add this deprecation, because we don't give any other working alternative in 3.4.